overlord/snapstate/snapstate.go

```diff
@@ -1244,20 +1244,28 @@ func Switch(st *state.State, name, channel string) (*state.TaskSet, error) {
 	return state.NewTaskSet(switchSnap), nil
 }
 
+type RevisionOptions struct {
```

Comment:
// RevisionsOptions control the selection of a snap revision. 

or something like that